text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,1678 @@
+    * Configure default registry: `npm config set registry https://registry.your-company.com/`
+    * Verify configuration: `npm config get registry`
+
+#### Advanced Options
+
+```
+# Generate only (skip dependency installation and git initialization)
+$ cdk init app --language=typescript --generate-only
+
+# Use a specific CDK library version
+$ cdk init app --language=typescript --lib-version 2.100.0
+
+# Combine both options
+$ cdk init app --language=typescript --generate-only --lib-version 2.100.0
+
+# Works with custom templates too
+$ cdk init --git-url https://github.com/user/my-template.git --generate-only
+$ cdk init --npm-package @myorg/template --lib-version 2.100.0
+```
+
+
+
+### Authoring Custom CDK Templates
+
+The requirements for a custom template are that it contains a CDK supported language subdirectory, a file of the same type inside that subdirectory, and the appropriate language-specific dependency file. Here are some example templates to reference: https://github.com/aws/aws-cdk-cli/tree/main/packages/aws-cdk/lib/init-templates
+
+
+#### Custom Template Schema:
+
+```
+my-custom-app-template/
+└── [language]/                          # REQUIRED: Language subdirectory (typescript, python, java, etc.)
+    ├── [at-least-one-file].[ext]        # REQUIRED: At least one file matching the language type in language subdirectory
+    ├── [dependency-file]                # REQUIRED: Language-specific dependency file
+    ├── cdk.template.json                # OPTIONAL: CDK configuration - required for app templates
+    ├── bin/                             # OPTIONAL: CDK app entry point directory
+    │   └── [entry-file].template.[ext]  # OPTIONAL: App entry file
+    ├── lib/                             # OPTIONAL: Stack definitions directory
+    │   └── [stack-file].template.[ext]  # OPTIONAL: Stack class file
+    ├── test/                            # OPTIONAL: Unit tests directory
+    │   └── [test-file].template.[ext]   # OPTIONAL: Test files
+    ├── [config-files]                   # OPTIONAL: Language-specific configuration files
+    ├── .template.gitignore              # OPTIONAL: Git ignore patterns
+    └── README.template.md               # OPTIONAL: Documentation
+```
+
+#### Example of CDK TypeScript App Template
+
+```
+my-custom-template/
+└── typescript/
+    ├── package.json               # Required dependency file
+    ├── cdk.template.json          # Required for app templates
+    ├── bin/
+    │   └── app.template.ts        # App entry file
+    ├── lib/
+    │   └── stack.template.ts      # Stack class file
+    ├── test/
+    │   └── stack.test.template.ts # Test file
+    ├── tsconfig.json              # TypeScript configuration
+    ├── .template.gitignore        # Git ignore patterns
+    └── README.md                  # Documentation
+```
+
+###### **Example Implementation of /package.json for TypeScript template**
+
+```
+{
+  "app": "npx ts-node --prefer-ts-exts bin/%name%.ts",
```

Comment:
This still uses place holders, which we said we won't be supporting for custom templates.